encode/src/main/scala/org/broadinstitute/monster/etl/encode/extract/EncodeIngestExtract.scala

```diff
@@ -0,0 +1,148 @@
+package org.broadinstitute.monster.etl.encode.extract
+
+import caseapp.HelpMessage
+import com.spotify.scio.ContextAndArgs
+import org.broadinstitute.monster.etl.encode.EncodeEntity
+import org.broadinstitute.monster.etl.encode.extract.client.EncodeClient
+import cats.effect.{ExitCode, IO, IOApp}
+import cats.implicits._
+import com.spotify.scio.extra.json._
+import com.spotify.scio.values.SCollection
+import io.circe.JsonObject
+import io.circe.syntax._
+
+object EncodeIngestExtract extends IOApp {
+
+  case class Args(
+    @HelpMessage("List of Assays")
+    assayTypes: List[String],
+    @HelpMessage("Output dir to JSON for ENCODE outputs")
+    outputDir: String
+  )
+
+  override def run(rawArgs: List[String]): IO[ExitCode] = {
+    val (pipelineContext, parsedArgs) = ContextAndArgs.typed[Args](rawArgs.toArray)
+
+    EncodeClient.resource.use { client =>
+      // for each entity, get the params and the extract based off of those params
+      val encodeExtractions = new EncodeExtractions(client)
+
+      // make assayTypes in scollection
+      val experimentsSearchParams = encodeExtractions.getExperimentSearchParams(
+        EncodeEntity.Experiment.entryName
+      )(pipelineContext.parallelize(parsedArgs.assayTypes))
+      val experiments = encodeExtractions.extractExperimentSearchParams(
+        EncodeEntity.Experiment.entryName,
+        EncodeEntity.Experiment.entityType
+      )(experimentsSearchParams)
+      experiments.saveAsJsonFile(
+        s"${parsedArgs.outputDir}/${EncodeEntity.Experiment.entryName}.json"
+      )
+
+      val fileIDParams = encodeExtractions.getIDParams(
+        EncodeEntity.File.entryName,
+        referenceField = "files",
+        manyReferences = true
+      )(experiments)
+      val filteredFiles = filterFiles(EncodeEntity.File.entryName)(
+        encodeExtractions.extractIDParamEntities(
+          EncodeEntity.File.entryName,
+          EncodeEntity.File.entityType
+        )(fileIDParams)
+      )
+      filteredFiles.saveAsJsonFile(
+        s"${parsedArgs.outputDir}/${EncodeEntity.File.entryName}.json"
+      )
+
+      val auditEntryName: String = "Audits"
+      val auditIDParams = encodeExtractions.getIDParams(
+        auditEntryName,
+        referenceField = "@id",
+        manyReferences = false
+      )(filteredFiles)
+      val transformedAudits = transformAudits(auditEntryName)(
+        encodeExtractions.extractIDParamEntities(auditEntryName, entityType = "File")(
+          auditIDParams
+        )
+      )
+      transformedAudits.saveAsJsonFile(s"${parsedArgs.outputDir}/$auditEntryName.json")
+
+      val replicateEntryName: String = "Replicates"
```

Comment:
I'd make new `case object`s in `EncodeEntity.scala` for Replicates & Audits (and anything else not already in there). We'll need them eventually for the transformation side of things.

You can just `???` the other methods for now.